includes/class-wc-stripe-api.php

```diff
@@ -233,6 +247,13 @@ public static function request( $request, $api = 'charges', $method = 'POST', $w
 	public static function retrieve( $api ) {
 		WC_Stripe_Logger::log( "{$api}" );
 
+		// If we have a transient indicating that the secret key is not valid, we dont't attempt the API call and we return an error.
+		$transient_key = WC_Stripe_Mode::is_test() ? self::TEST_MODE_INVALID_API_KEYS_TRANSIENT_KEY : self::LIVE_MODE_INVALID_API_KEYS_TRANSIENT_KEY;
+		$invalid_api_keys_detected = get_transient( $transient_key );
+		if ( $invalid_api_keys_detected ) {
+			return new WP_Error( 'stripe_error', __( 'The Stripe API keys are not valid.', 'woocommerce-gateway-stripe' ) );
+		}
+
```

Comment:
I think we should do this check before we trigger the logger, otherwise we're going to flood the logs.
```suggestion
		// If we have a transient indicating that the secret key is not valid, we don't attempt the API call and we return an error.
		$transient_key = WC_Stripe_Mode::is_test() ? self::TEST_MODE_INVALID_API_KEYS_TRANSIENT_KEY : self::LIVE_MODE_INVALID_API_KEYS_TRANSIENT_KEY;
		$invalid_api_keys_detected = get_transient( $transient_key );
		if ( $invalid_api_keys_detected ) {
			return new WP_Error( 'stripe_error', __( 'The Stripe API keys are not valid.', 'woocommerce-gateway-stripe' ) );
		}

		WC_Stripe_Logger::log( "{$api}" );

```